### PR TITLE
Projectadded

### PR DESCRIPTION
**Pull request by Usagib**
Pull request for code review

### CODE REVIEW

**Comment by Usagib** [comment]
Hi team!
I have made a new repo and added gitflow guidelines, here is my first project repo: https://github.com/Usagib/rubyenumerable/
Happy coding!

File: script.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+# enumerable ruby module
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+  end
```
Comment (imhta) [inline script.rb]: In `my_each`
- my_each returns an Enumerator if no block is given

File: script.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
```
Comment (imhta) [inline script.rb]: In `my_each_with_index`
- my_each_with_index returns an enumerator if no block is given

File: script.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  def my_select
+    i = 0
+    res_arr = []
+    length.times do
+      res_arr << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    res_arr
+  end
```
Comment (imhta) [inline script.rb]: In `my_select`
- my_select returns an enumerator if no block is given

File: script.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  def my_all?
+    i = 0
+    length.times do
+      return false if yield(self[i]) == false
+
+      i += 1
+    end
+    true
+  end
```
Comment (imhta) [inline script.rb]: In `my_all`
- my_all? when no block or argument is given returns true when none of the collection members are false or nil
- my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: script.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  def my_none?
+    i = 0
+    length.times do
+      return false if yield(self[i]) == true
+
+      i += 1
+    end
+    true
+  end
+
```
Comment (imhta) [inline script.rb]: In `my_none?`
-  my_none? when no block or argument is given returns true only if none of the collection members is true
- my_none? when a class is passed as an argument returns true if none of the collection is 
a member of such class
- my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
-  my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: script.rb [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  def my_count
+    i = 0
+    c = 0
+    length.times do
+      c += 1 if yield(self[i]) == true
+      i += 1
+    end
+    c
+  end
```
Comment (imhta) [inline script.rb]: In `my_count`
- my_count returns the number of items in enum through enumeration
- my_count counts the number of items in enum that are equal to item if an argument is given

File: script.rb [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  def my_map
+    i = 0
+    res_array = []
+    length.times do
+      res_array << yield(self[i]) if block_given?
+      res_array << proc.call(self[i])
+      i += 1
+    end
+    res_array
+  end
```
Comment (imhta) [inline script.rb]: In `my_map`
- my_map returns a new array with the results of running block once for every element in enum.
- my_map returns an Enumerator if no block is given

File: script.rb [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+  def my_inject(start_value)
+    my_each { |element| start_value = yield(start_value, element) }
+    start_value
+  end
+end
```
Comment (imhta) [inline script.rb]: In `my_inject`
- my_inject combines all elements of enum by applying a binary operation, specified by a block
- my_inject when a symbol is specified combines each element of the collection by applying 
the symbol as a named method

**Review by imhta — CHANGES_REQUESTED** [review]
# Require changes 🛑 

### Hello @Usagib , 👋  There are some test are failing in you methods. kindly check the inline comment. and `my_any` method is missing. for reference [link](https://ruby-doc.org/core-2.6.5/Enumerable.html)

----
#### Don't forgot to request for another code review
----

**Pon Muthu Selvam N**
_If you have any queries, feel free to ping me on slack_

**Pull request closed without merging**